Source/WTF/wtf/FileSystem.cpp

```diff
@@ -909,7 +909,7 @@ String createTemporaryDirectory()
 
     std::string newTempDirTemplate = tempDir + "XXXXXXXX";
 
-    Vector<char> newTempDir(newTempDirTemplate.c_str(), newTempDirTemplate.size());
+    Vector<char> newTempDir(std::span<const char> { newTempDirTemplate.data(), newTempDirTemplate.size() });
```

Comment:
Yes, we actually convert std::string into a span several times in this patch. I'll look into adding a convenience function to StdLibExtras.h. Would `std::span<const char> toSpan(const std::string&)` sound OK?

Reply:
It’s OK with me.

I'm not sure how we are deciding when to mention the names of arguments. I’d love to have a `toSpan` or `span` function overloaded to work with lots of different types. But I noticed that we have functions with names like `spanFromNSData` instead of just an overloaded function. And member functions named `span` in other cases. That’s all OK but a broader pattern might make this nicer.

Reply:
Oh, maybe I should rename `spanFromNSData` to `toSpan` too then. Having overloaded functions seems good to me and `toSpan` is nice and short.

Reply:
Oh, there is no `spanFromNSData`. I am guessing this was just an example. We may well need such a function though.

Reply:
I confused myself. I suggested `spanFromNSData`, but it doesn’t exist yet. But same may apply for `vectorFromNSData`/`toVector`.